### PR TITLE
[Backend/Apps] Documentation

### PR DESCRIPTION
**Pull request by viacheslav-huma**
This PR is intended to gather all the documentations from backend/apps/web in one place before merging to the master

### CODE REVIEW

File: data-collection/backend/client-integration/index.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+print(response)
+
+# Login a user
+response = client.login("user@example.com")
+print(response)
+```
+
+### Endpoint Implementation Example
+
+User authentication should be handled via your original endpoints. The "Get Huma Token" endpoint (`{domain}/private/huma-token`) should be protected and based on the current authenticated user, perform the requests to Huma backend.
```
Comment (mohsen-huma) [inline data-collection/backend/client-integration/index.md:56]:
```suggestion
User authentication should be handled via your original endpoints. The "Get Huma Token" endpoint (for example: `{domain}/private/huma-token`) should be protected and based on the current authenticated user, perform the requests to Huma backend.
```

File: data-collection/backend/client-integration/index.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+def get_huma_token(request_body: HumaTokenRequestBody = Depends(), current_user: User = Depends(authenticate)) -> HumaTokenResponse:
+    """
+    Interacts with Huma backend to auth the user and provide the auth and refresh tokens for Huma sdk.
+    """
+    huma_api = HumaApiClient(config_file="huma-config.json")
+    try:
+        rsp_json = huma_api.login(current_user.email, request_body.client_type)
+    except HumaError:
+        rsp_json = sign_up_user(current_user, huma_api, request_body.client_type)
+    return HumaTokenResponse(
```
Comment (mohsen-huma) [inline data-collection/backend/client-integration/index.md:90]: No need to make an object. JSON is enough.

File: data-collection/backend/client-integration/index.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    )
+
+
+def sign_up_user(user: User, huma_api: HumaApiClient, client_type: HumaClientType) -> dict:
+    try:
+        huma_api.register(user.email, user.firstName, user.lastName, client_type)
+    except HumaError as e:
+        raise HTTPException(status_code=400, detail=f"Huma signup error: {str(e)}")
+    return huma_api.login(user.email, client_type)
+
```
Comment (mohsen-huma) [inline data-collection/backend/client-integration/index.md:105]:
```suggestion
from fastapi import APIRouter, Depends
from huma_python_client.client import HumaApiClient, APIClientError as HumaNonExistantUserError

from app.dependencies import authenticate
from app.models.database import User
from app.models.user import HumaTokenRequestBody

router = APIRouter(prefix="/private", dependencies=[Depends(authenticate)], tags=['Huma'])

@router.get("/huma-token")
def get_huma_token(request_body: HumaTokenRequestBody = Depends(), user: User = Depends(authenticate)) -> HumaTokenResponse:
    """
    Interacts with Huma backend to auth the user and provide the auth and refresh tokens for Huma sdk.
    """
    huma_client = HumaApiClient(config_file_path="huma-config.json")
    try:
        json_response = huma_client.login(user.email, request_body.client_type)
    except HumaNonExistantUserError:
        json_response = huma_client.register(user.email, user.firstName, user.lastName, request_body.client_type)
    return json_response

```

File: data-collection/backend/client-integration/index.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+```
```
Comment (mohsen-huma) [inline data-collection/backend/client-integration/index.md:106]:
```suggestion
```py
from fastapi import APIRouter, Depends
from huma_python_client.client import HumaApiClient, APIClientError as HumaNonExistantUserError

from app.dependencies import authenticate
from app.models.database import User
from app.models.user import HumaTokenRequestBody

router = APIRouter(prefix="/private", dependencies=[Depends(authenticate)], tags=['Huma'])

@router.get("/huma-token")
def get_huma_token(request_body: HumaTokenRequestBody = Depends(), user: User = Depends(authenticate)) -> HumaTokenResponse:
    """
    Interacts with Huma backend to auth the user and provide the auth and refresh tokens for Huma sdk.
    """
    huma_client = HumaApiClient(config_file_path="huma-config.json")
    try:
        json_response = huma_client.login(user.email, request_body.client_type)
    except HumaNonExistantUserError:
        json_response = huma_client.register(user.email, user.firstName, user.lastName, request_body.client_type)
    return json_response

```
```

**Review by mohsen-huma — APPROVED** [review]

**Review by jaygajjar-huma — APPROVED** [review]

**Review by milanof-huma — APPROVED** [review]

**Pull request merged**